### PR TITLE
fix for tcs's RunContinuationsAsynchronously

### PR DESCRIPTION
**Pull request by realwanpengli**

### CODE REVIEW

**Comment by vicancy** [comment]
Adding a ut for this.

File: src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ internal abstract class ServiceConnectionBase : IServiceConnection
 
         private readonly SemaphoreSlim _serviceConnectionLock = new SemaphoreSlim(1, 1);
 
-        private readonly TaskCompletionSource<bool> _serviceConnectionStartTcs = new TaskCompletionSource<bool>(TaskContinuationOptions.RunContinuationsAsynchronously);
+        private readonly TaskCompletionSource<bool> _serviceConnectionStartTcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
```
Comment (scale-tone) [inline src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs]: So, where are unit tests for this?

As we can now see, the code here is extremely fragile (confusing a ctor, that caused a critical hidden bug is a strong evidence of that). How can it **still** not be covered by tests?

**Comment by realwanpengli** [comment]
appveyor always fail, wierd.
see #639

File: src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs (hunk 1 above)
Comment (realwanpengli) [reply src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs]: Added. this bug only occurs in console app.

**Pull request merged**